src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -67,6 +67,16 @@ AndroidDeviceControllerWrapper::~AndroidDeviceControllerWrapper()
         delete mDeviceAttestationDelegateBridge;
         mDeviceAttestationDelegateBridge = nullptr;
     }
+    if (mDeviceAttestationVerifier != nullptr)
+    {
+        delete mDeviceAttestationVerifier;
+        mDeviceAttestationVerifier = nullptr;
+    }
+    if (mAttestationTrustStoreBridge != nullptr)
+    {
+        delete mAttestationTrustStoreBridge;
+        mAttestationTrustStoreBridge = nullptr;
+    }
```

Comment:
This assumes it was `new'ed` which it doesn't have to be. We should not delete unless we know it was allocated dynamically with new. 

The pattern used by previous code:
```
if (mDeviceAttestationDelegateBridge != nullptr)
    {
        delete mDeviceAttestationDelegateBridge;
        mDeviceAttestationDelegateBridge = nullptr;
    }
```

is also wrong.

Reply:
I agree with that, using `std::unique_ptr` could make it clearer.

However, it requires all platforms `AttestationTrustStore *` and `DeviceAttestationVerifier *` to be modified into smart pointers. 

Could we fix it in following PR?

Reply:
Right now the issue is that you force a new being done outside, which is not guaranteed to be done by all...

Reply:
OK，I will fix it. Due to the need to modify the code for multiple platforms, I need to work with other teams in my company, which will take some time.

Reply:
Thanks @panliming-tuya